lib/seattleflu/db/cli/command/redcap_det.py

```diff
@@ -0,0 +1,180 @@
+
+    instruments = get_project_instruments(api_url, api_token)
+    redcap_records = get_redcap_records(api_url, api_token, start_date)
+
+    for record in redcap_records:
+        # Find all instruments within a record that have been mark completed
+        for instrument in instruments:
+            if record[(instrument + '_complete')] == '2':
+                print(as_json(create_det_records(redcap_url, project_id, record, instrument)))
+
+
+def get_project_instruments(api_url: str, api_token: str) -> List[str]:
+    """
+    Get REDCap instruments for a given project, which is determined by the
+    *api_token*
+    """
+    parameters = {
+        'content': 'instrument',
+        'format': 'json',
+        'token': api_token
+    }
+
+    instruments = get_redcap_data(api_url, parameters)
+    instrument_names = []
+
+    for instrument in instruments:
+        instrument_names.append(instrument['instrument_name'])
+
+    return instrument_names
+
+
+def get_redcap_records(api_url: str, api_token: str, start_date: str) -> List[dict]:
+    """
+    Get REDCap records for a given project, which is determined by the
+    *api_token*
+    """
+    parameters = {
+        'content': 'record',
+        'format': 'json',
+        'type': 'flat',
+        'token': api_token,
+        'dateRangeBegin': start_date
+    }
+
+    records = get_redcap_data(api_url, parameters)
+    return records
+
+
+def get_redcap_data(api_url: str, parameters: dict) -> List[dict]:
+    """
+    Get REDCap data by POST request to the REDCap API.
+
+    *api_url* specifies instance of REDCap and `api_token` within *parameters*
+    specifies the project.
+
+    Consult REDCap API documentation for required and optional parameters
+    to include in API request.
+    """
+    headers = {
+        'Content-type': 'application/x-www-form-urlencoded',
+        'Accept': 'application/json'
+    }
+
+    response = requests.post(api_url, data=parameters, headers=headers)
+
+    if response.status_code != 200:
+        raise Exception(f"REDCap returned response status code {response.status_code}")
+
+    return response.json()
+
+
+def create_det_records(redcap_url: str, project_id: str,
+                       record: dict, instrument: str) -> dict:
+    """
+    Create a "fake" DET record that matches the format of REDCap DETs:
+
+    \b
+    {
+        'redcap_url',
+        'project_id',
+        'record',
+        'instrument',
+        '<instrument>_complete',
+        'redcap_event_name' (for longitudinal projects only)
+    }
+    """
+    instrument_complete = instrument + '_complete'
+
+    det_record = {
+       'redcap_url': redcap_url,
+       'project_id': project_id,
+       'record': record['record_id'],
+       'instrument': instrument,
+       instrument_complete: record[instrument_complete]
+    }
+
+    if 'redcap_event_name' in record:
+        det_record['redcap_event_name'] = record['redcap_event_name']
```

Comment:
I haven't see these show up in the real DETs but that might be because we aren't receiving any from the Household studies. Not sure if they come in through those projects

Reply:
Looks like they do. I checked the kiosks project and the one-week followup is a repeating instrument. I pushed one more commit to add these fields.